### ISSUE TITLE
X11 apps can't be opened using xwayland

### ISSUE THREAD

**Issue by illode** [wez/wezterm#3055]
### What Operating System(s) are you seeing this problem on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

KDE Plasma/KWin

### WezTerm version

20230205-063357-20c8135f

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

x11 applications launched from wezterm crash immediately. The `enable_wayland` setting works for wezterm, but regardless of what it's set to it can't launch x11 porgrams. They launch fine from anywhere else, and launching wayland programs also works with no issue.

xeyes, for example:
```
❯ xeyes
Authorization required, but no authorization protocol specified

Error: Can't open display: :1
```

### To Reproduce

1. Open wezterm with/without wayland enabled
2. Attempt to launch an app that would open with xwayland

### Configuration

no config

### Expected Behavior

X11 programs open using xwayland.

### Logs

There are no logs:
```
❯ cat $XDG_RUNTIME_DIR/wezterm/*log*
zsh: no matches found: /run/user/1000/wezterm/*log*
```

**Comment by wez** [comment]
Sounds like an issue with your environment; check your shell startup files to see if they are either not setting things up, or are clearing things out of the environment that might affect this. You can google that error message to get some idea of `xhost` and related X11 authorization configuration to help you track down what might be going on with this.

Note that wezterm launches a login shell by default, and some users configure their environment differently in a login shell.

**Comment by illode** [comment]
Fixed, was an issue on my end.

I didn't realize wezterm used a login shell while the other terminals I tried didn't. That made the problem much more obvious:

I globally set XDG vars, as well as some some vars in `/etc/profile.d/local-set-xdg-vars.sh` so I can keep $HOME clean using [this](https://wiki.archlinux.org/title/XDG_Base_Directory) as a reference.

This export was causing the issue:
```
export XAUTHORITY="$XDG_RUNTIME_DIR"/Xauthority
```
It was overriding the path to the actual xauth file, which is partially random and would change each session, e.g. `$XDG_RUNTIME_DIR/xauth_NKMqFo`.

I assume it's set by the WM when it starts the xwayland server, hence it doesn't use the usual Xauthority name.
